### ISSUE TITLE
Support macOS universal2 wheel fusion?

### ISSUE THREAD

**Issue by henryiii** [pypa/cibuildwheel#705]
In the future, would it make sense to support macOS universal2 wheel fusion, like [multibuild does](https://github.com/matthew-brett/multibuild/pull/383)? Either automatically when all three arches are chosen, for example?

**Comment by mayeut** [comment]
I'm +1 on this, that was one workflow that made sense to me https://github.com/pypa/cibuildwheel/issues/473#issuecomment-748471250
From the link you provided to multibuild, seems delocate already support this which is good news (I didn't check back in December that it was the case).

**Comment by mattip** [comment]
FWIW, multibuild is backtracking and will also create thin wheels. A problem was [discovered with universal2](https://github.com/MacPython/numpy-wheels/issues/140) and NPY_SIZEOF_LONGDOUBLE and NPY_SIZEOF_COMPLEX_LONGDOUBLE for numpy, which was fixed in numpy/numpy#20270, but the problem may appear in other projects so it is safer to distribute separate wheels.

**Comment by isuruf** [comment]
> FWIW, multibuild is backtracking and will also create thin wheels.

Not true. multibuild has always created thin wheels and did wheel fusion.

**Comment by henryiii** [comment]
Yes, this is not related. NumPy is just shipping the thin wheels now. This issue is about wheel fusion, that is, making the thin wheels and then post processing them into a universal wheel, specifically making sure it's easy to do that combination step.

**Comment by mattip** [comment]
As I commented [elsewhere](https://github.com/numpy/numpy/pull/20787#issuecomment-1068815626) I don't really see the need for cibuildwheel to provide this option. I haven't heard a clear need for `pip install <universal2wheel>`, and for app packagers I think the correct place to fuse wheels is when creating the application. Please correct me if I am mistaken.
